polar-core/src/counter.rs

```diff
@@ -0,0 +1,51 @@
+use std::sync::atomic::{AtomicU64, Ordering};
```

Comment:
No, this is factoring the counter aspect of the KB into its own first-class thing so that we can use them in runnables without passing around the whole KB. The behavior of the counter is the same as before (e.g., wrapping at 2**53).